lib/validators.test.ts

```diff
@@ -0,0 +1,134 @@
+import { generateFlexibleSchema } from './validators';
+
+describe('generateFlexibleSchema', () => {
+  it('handles different field types', async () => {
+    const schema = generateFlexibleSchema([
+      {
+        question: 'foo',
+        id: 'one',
+        type: 'text',
+      },
+      {
+        question: 'foo',
+        id: 'two',
+        type: 'textarea',
+      },
+      {
+        question: 'foo',
+        id: 'three',
+        type: 'checkboxes',
+      },
+      {
+        question: 'foo',
+        id: 'four',
+        type: 'repeater',
+      },
+      {
+        question: 'foo',
+        id: 'five',
+        type: 'select',
+      },
+      {
+        question: 'foo',
+        id: 'six',
+        type: 'radios',
+      },
+    ]);
+
+    const result = await schema.validate({
+      one: 'value',
+      two: 'value',
+      three: ['example', 'example 2'],
+      four: ['example', 'example 2'],
+      five: 'value',
+      six: 'value',
+    });
+
+    expect(result).toBeTruthy();
+  });
+
+  it('handles required fields', async () => {
```

Comment:
can we see a test for require passing

Reply:
i'm not sure what you mean. you mean a test case where it doesn't throw an error?

Reply:
Yes so that when a field is required and we give it a response no errors are thrown.

Reply:
done